index.js

```diff
@@ -339,6 +339,64 @@ app.post('/api/item/cancel-reservation/:id', async (req, res) => {
   }
 });
 
+
+// POST /api/item/return/:id endpoint
+app.post('/api/item/return/:id', async (req, res) => {
+  const id = parseInt(req.params.id);
+  const user_id = parseInt(req.query.user_id);
+
+  if (isNaN(id) || isNaN(user_id)) {
+    return res.status(400).send({
+      error: "BAD_REQUEST",
+      message: "please provide a numeric user ID and item ID"
+    });
+  }
+
+  try {
+    // Select the resource that should be returned
+    const [resource] = await sql`
+      select id, owned_by, reserved_by, reservation_status
+      from resources
+      where id = ${id};
+    `;
+
+    if (!resource) { // No resource found
+      return res.status(404).send({
+        error: API_RETURN_MESSAGES.ITEM_UNAVAILABLE,
+        id
+      });
+    } else if (resource.reservation_status != ITEM.CONFIRMED) { // The resource must be confirmed as borrowed
+      return res.status(403).send({
+        error: API_RETURN_MESSAGES.NOT_BORROWED,
+        message: "Cannot return an item that hasn't been borrowed."
+      });
+    } else if (resource.reserved_by != user_id) { // The user cancelling the reservation has to be the borrower
+      return res.status(401).send({
+        error: API_RETURN_MESSAGES.UNAUTHORIZED,
+        message: "You are not authorized to perform this action."
+      });
+    }
+
+    const _update_result = await sql`
+      update resources
+      set reservation_status = ${ITEM.UNLISTED}, reserved_by = NULL where id = ${id};
```

Comment:
Looks good but is there a reason the item is set as unlisted after it's returned?

Reply:
I figured we shouldn't assume the owner wants to lend it out again, though maybe a `relist` bool query param would be better?

Reply:
That sounds good, but you have a point. Listing/unlisting should be at the owner's discretion. Maybe that should be a separate feature?

Reply:
True, I don't think we have an endpoint specifically for listing and unlisting resources right now - probably worth creating an issue for that